### ISSUE TITLE
[Feature Request] Make a Distinction Between Loading and Error States in Image Component

### ISSUE THREAD

**Issue by Arian94** [nextui-org/nextui#3640]
### Is your feature request related to a problem? Please describe.

When using the Image component, one would want to have two different images for the _loading_ state as well as _showing a fallback image_ when an error is encountered.

ATM, there is only one property called ```fallbackSrc``` that handles both states and displays one image, so there is no distinction between the loading and the error states.

### Describe the solution you'd like

Please, make it possible to display another image on the Error state by adding a new property and modifying the ```use-image.ts``` file to handle such case.

### Describe alternatives you've considered

Wrap the existing Image component and on the ```onError``` handler, set the ```fallbackSrc``` to point to another source like:
```
const [fallbackSrc, setFallbackSrc] = useState('');

<Image
    fallbackSrc={fallbackSrc}
    onError={() => {
         setFallbackSrc('/assets/another-img');
     }}
 ...
/>
```

**Comment by abhinav700** [comment]
@wingkwong can you assign this issue to me please

**Comment by kaizerpwn** [comment]
I’ve encountered the same issue recently and I completely agree with you.

**Comment by abhinav700** [comment]
@wingkwong Can you assign this issue to me. I would like to work on it.

**Comment by Arian94** [comment]
@wingkwong Oh, wasn't used to that. From my experience, the quicker and better you do, the higher chances of success.
But alright, got it :)

**Comment by wingkwong** [comment]
I agree but you may at least leave a message saying you will be working on it. Otherwise I'll assume no one is working on it and assign to someone else. This would waste the effort and time for others. I'll check both PR to see which one suits better and close the other one.

cc @abhinav700

**Comment by Arian94** [comment]
> I agree but you may at least leave a message saying you will be working on it. Otherwise I'll assume no one is working on it and assign to someone else. This would waste the effort and time for others. I'll check both PR to see which one suits better and close the other one.
> 
> cc @abhinav700

Yeah, it happened so fast.

**Comment by abhinav700** [comment]
> I'll check both PR to see which one suits better and close the other one.
>
> cc @abhinav700

Okay.

**Comment by wingkwong** [comment]
@Arian94 Can you recreate the PR again?

**Comment by Arian94** [comment]
> [@Arian94](https://github.com/Arian94) Can you recreate the PR again?

I'll try.

**Comment by wingkwong** [comment]
Closing in favor of v3. If the issue is reproducible in v3, please create a new one with a minimal reproducible example.
